test/integration/eth-simulator.ts

```diff
@@ -81,26 +80,67 @@ test('ETH: send allowed eth63', async t => {
       t.end()
     }
   }
-  util.twoPeerMsgExchange(t, capabilities, opts)
+  util.twoPeerMsgExchange(t, opts, cap)
+}
+
+test('ETH: should use latest protocol version on default', async t => {
+  sendWithProtocolVersion(t, 64)
 })
 
-test('ETH: send allowed eth62', async t => {
-  const cap = [devp2p.ETH.eth62]
+test('ETH: should work with allowed eth64', async t => {
+  sendWithProtocolVersion(t, 64)
+})
+
+test('ETH -> Eth64 -> sendStatus(): should throw on non-matching latest block provided', async t => {
+  const cap = [devp2p.ETH.eth64]
+  const common = new Common({ chain: 'mainnet', hardfork: 'byzantium' })
+  const status0: any = Object.assign({}, status)
+  status0['latestBlock'] = 100000 // lower than Byzantium fork block 4370000
+
+  const rlpxs = util.initTwoPeerRLPXSetup(null, cap, common)
+  rlpxs[0].on('peer:added', function(peer: any) {
+    const protocol = peer.getProtocols()[0]
+    t.throws(() => {
+      protocol.sendStatus(status0)
+    }, /latest block provided is not matching the HF setting/)
+    util.destroyRLPXs(rlpxs)
+    t.end()
+  })
+})
+
+test('ETH -> Eth64 -> ForkId validation 1a)', async t => {
   const opts: any = {}
-  opts.status0 = Object.assign({}, status)
+  const cap = [devp2p.ETH.eth64]
+  const common = new Common({ chain: 'mainnet', hardfork: 'byzantium' })
+  const status0: any = Object.assign({}, status)
+  // Take a latest block > next mainnet fork block (constantinople)
+  // to trigger validation condition
+  status0['latestBlock'] = 9069000
+  opts.status0 = status0
   opts.status1 = Object.assign({}, status)
-  opts.onOnceStatus0 = function(rlpxs: any, eth: any) {
-    eth.sendMessage(devp2p.ETH.MESSAGE_CODES.NEW_BLOCK_HASHES, [437000, 1, 0, 0])
-    t.pass('should send NEW_BLOCK_HASHES message')
-  }
-  opts.onOnMsg1 = function(rlpxs: any, eth: any, code: any) {
-    if (code === devp2p.ETH.MESSAGE_CODES.NEW_BLOCK_HASHES) {
-      t.pass('should receive NEW_BLOCK_HASHES message')
-      util.destroyRLPXs(rlpxs)
-      t.end()
-    }
+  opts.onPeerError0 = function(err: Error, rlpxs: any) {
+    const msg = 'Remote is advertising a future fork that passed locally'
```

Comment:
We might want to create some kind of global error-string object, such that this test doesn't suddenly fail if we change the error string. We can then test if the msg is of a certain `Error` (an enum or something just like we did in VM). (Not a blocker)

Reply:
Yeah, that's a good idea, @rumkin had also already been working on a similar proposal on the VM https://github.com/ethereumjs/ethereumjs-vm/issues/487 (that one is further going though - just wanted to mention here. For a first round it definitely makes sense to just start with the ERROR strings).

Reply:
Yes, I agree, this is something we should do midterm, or at the point where these "magic strings" are in "too many" places.